test/enigma_test.rb

```diff
@@ -18,46 +16,124 @@ def test_it_has_attributes
+    @enigma.stubs(:default_key).returns("02715")
+    expected = {
+      decryption: "hello world",
+      key: "02715",
+      date: "040895"
+    }
+    assert_equal expected, @enigma.decrypt("keder ohulw")
   end
-end
 
-# def test_it_can_split_return_value_of_generate_random_key_into_separate_keys
-#   expected = { :key_A => 02, :key_B => 27, :key_C => 71, :key_D => 15}
-#   assert_equal expected, @shift_generator.separate_random_key("02715")
-#
-#   random_test_key = @shift_generator.generate_random_key
-#   assert_instance_of Hash, @shift_generator.separate_random_key(random_test_key)
-#   assert_equal 4, @shift_generator.separate_random_key(random_test_key).values.length
-# end
-#
-# def test_it_can_square_date_parameter
-#   test_date = "270220"
-#   assert_equal "73018848400", @shift_generator.square_date(test_date)
-# end
-#
-# def test_it_can_get_last_4_digits_from_sqaure_date
-#   squared_date_param = "73018848400"
-#   assert_equal "8400", @shift_generator.last_four_digits(squared_date_param)
-# end
-#
-# def test_it_can_separate_last_4_digits_from_square_dates_into_keys
-#   four_digits = "8400"
-#   expected = {key_A: 8, key_B: 4, key_C: 0, key_D: 0}
-#   assert_equal expected, @shift_generator.separate_last_4_digits(four_digits)
-# end
-#
-# def test_it_can_get_final_shift_values
-#   expected = {key_A: 10, key_B: 31, key_C: 71, key_D: 15}
-#   assert_equal expected, @shift_generator.final_shift_values("02715", "8400")
-# end
+  def test_it_can_decrypt_a_message_with_given_date_and_given_key
+    expected = {
+      decryption: "hello world",
+      key: "02715",
+      date: "270220"
+    }
+
+    assert_equal expected, @enigma.decrypt("rib ydmcapu", "02715", "270220")
+  end
+
+  def test_it_can_decrypt_a_message_with_given_key
+    @enigma.stubs(:default_date).returns("270220")
+
+    expected = {
+      decryption: "hello world",
+      key: "02715",
+      date: "270220"
+    }
+
+    assert_equal expected, @enigma.decrypt("rib ydmcapu", "02715")
+  end
+
+  def test_it_can_decrypt_a_message_with_given_date
+    @enigma.stubs(:default_key).returns("02715")
+
+    expected = {
+      decryption: "hello world",
+      key: "02715",
+      date: "270220"
+    }
+
+    assert_equal expected, @enigma.decrypt("rib ydmcapu", "270220")
+  end
+
+  def test_it_can_encrypt_a_message_with_special_characters
+    @enigma.stubs(:default_date).returns("040895")
+    @enigma.stubs(:default_key).returns("02715")
+    expected = {
+      encryption: "keder ohulw!",
+      key: "02715",
+      date: "040895"
+    }
+    assert_equal expected, @enigma.encrypt("hello world!")
+  end
+
+  def test_it_can_decrypt_a_message_with_default_values
+    @enigma.stubs(:default_date).returns("040895")
+    @enigma.stubs(:default_key).returns("02715")
+    expected = {
+      decryption: "hello world!",
+      key: "02715",
+      date: "040895"
+    }
+    assert_equal expected, @enigma.decrypt("keder ohulw!")
+  end
+end
```

Comment:
tried to test as many edge cases as I could think of, one issue is that not sure if special characters should take a shift index spot.  Not sure how to implement that functionality though, will try to add a guard clause